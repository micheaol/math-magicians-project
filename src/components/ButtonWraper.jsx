/* eslint-disable linebreak-style */
/* eslint-disable react/prop-types */
/* eslint-disable arrow-body-style */
import './ButtonWraper.css';
import React from 'react';

const ButtonWraper = ({ children }) => {
  return (
    <div className="buttonBox">{children}</div>
  );
};

export default ButtonWraper;
