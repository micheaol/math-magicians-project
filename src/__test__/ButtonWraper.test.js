import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ButtonWraper from '../components/ButtonWraper';

afterEach(() => {
  cleanup();
});

const btnValues = [
  ['AC', '+/-', '%', 'รท'],
  [7, 8, 9, 'x'],
  [4, 5, 6, '-'],
  [1, 2, 3, '+'],
  [0, '.', '='],
];

describe('render component to screen', () => {
  test('render ButtonWraper to the screen', () => {
    render(<ButtonWraper>{btnValues}</ButtonWraper>);
    const buttonWrapperElement = screen.getByTestId('button-wraper');
    expect(buttonWrapperElement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<ButtonWraper>{btnValues}</ButtonWraper>);
    expect(tree).toMatchSnapshot();
  });
});
