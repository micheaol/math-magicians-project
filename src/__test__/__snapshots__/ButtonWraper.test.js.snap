// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component to screen matches snapshot 1`] = `
<div
  className="buttonBox"
  data-testid="button-wraper"
>
  AC
  +/-
  %
  รท
  7
  8
  9
  x
  4
  5
  6
  -
  1
  2
  3
  +
  0
  .
  =
</div>
`;
